lib/CL/devices/cuda/pocl-cuda.c

```diff
@@ -246,15 +250,21 @@ pocl_cuda_init_device_infos (struct _cl_device_id *dev)
   dev->has_64bit_long = 1;
 }
 
+#define CUDA_PROBE_CHECK(ret) if (ret != CUDA_SUCCESS) return 0;
+
 unsigned int
 pocl_cuda_probe (struct pocl_device_ops *ops)
 {
   int env_count = pocl_device_get_env_count (ops->device_name);
 
-  // TODO: Check how many CUDA device available (if any)
-
   if (env_count < 0)
-    return 1;
+    {
+      CUresult ret = cuInit(0);
```

Comment:
This needs to be before the conditional block, otherwise if we run with `POCL_DEVICES=CUDA` then `cuInit` is never called.

Also use GNU C style (space before `(`) - same for next 3 lines.